scripts/manage-db.py

```diff
@@ -118,34 +120,61 @@ def extract_active_data(url, dump_location='dump.sql'):
-        rules_scheduled_changes_history migrate_version  > %s' % (mysql_default_command, database, dump_location,))
+        rules_scheduled_changes_history migrate_version  > %s' % (mysql_default_command, database, dump_location))
     )
 
     popen(
         _strip_multiple_spaces('%s %s releases --where="EXISTS ( \
             SELECT * \
             FROM rules, rules_scheduled_changes \
-            WHERE releases.name = rules.mapping \
-               OR releases.name = rules.whitelist \
-               OR releases.name = rules_scheduled_changes.base_mapping \
-               OR releases.name = rules_scheduled_changes.base_whitelist \
+            WHERE releases.name IN ( \
+                rules.mapping, \
+                rules.whitelist, \
+                rules.fallbackMapping, \
+                rules_scheduled_changes.base_mapping, \
+                rules_scheduled_changes.base_whitelist, \
+                rules_scheduled_changes.base_fallbackMapping) \
             )" \
-        >> %s' % (mysql_default_command, database, dump_location,))
+        >> %s' % (mysql_default_command, database, dump_location))
     )
 
     popen(
         _strip_multiple_spaces('%s %s releases_history --where="releases_history.name=\'Firefox-mozilla-central-nightly-latest\' \
-        AND 1 limit 50" >> %s' % (mysql_default_command, database, dump_location,))
+        limit 50" >> %s' % (mysql_default_command, database, dump_location))
     )
 
     popen(
         _strip_multiple_spaces('%s --skip-add-drop-table --no-create-info %s releases_history --where "name = ( \
             SELECT rules.mapping \
             FROM rules \
             WHERE rules.alias=\'firefox-release\' \
-        ) LIMIT 50" >> %s' % (mysql_default_command, database, dump_location,))
+        ) LIMIT 50" >> %s' % (mysql_default_command, database, dump_location))
     )
 
+    query_release_mapping = """SELECT DISTINCT releases.* \
+        FROM releases, rules, rules_scheduled_changes \
+        WHERE releases.name IN ( \
+            rules.mapping, \
+            rules.whitelist, \
+            rules.fallbackMapping, \
+            rules_scheduled_changes.base_mapping, \
+            rules_scheduled_changes.base_whitelist, \
+            rules_scheduled_changes.base_fallbackMapping \
+            )"""
+
+    result = trans.execute(query_release_mapping).fetchall()
+    partial_release_names = set()
+    for row in result:
+        try:
+            release_blob = createBlob(row['data'])
+            partial_release_names = partial_release_names.union(release_blob.getReferencedReleases())
+        except:
+            continue
+    if partial_release_names:
+        qry = ", ".join("'" + release_names + "'" for release_names in partial_release_names)
+        popen(_strip_multiple_spaces('%s %s releases --where="releases.name IN (%s)" \
+               >> %s' % (mysql_default_command, database, qry, dump_location)))
```

Comment:
I think this will be okay, but there's a small possibility that we may need to batch this if the queries or shell commands end up too long. We'll see what happens when we run it against the production db.